sdk/evm/executor.go

```diff
@@ -108,3 +145,103 @@ func (e *Executor) SetRoot(
 		RawData:     tx,
 	}, err
 }
+
+// buildExecuteCallData packs calldata for ManyChainMultiSig.execute(...)
+func buildExecuteTxData(
+	opts *bind.TransactOpts,
+	mcmsAddr common.Address,
+	op bindings.ManyChainMultiSigOp,
+	proof [][32]byte,
+) (*gethtypes.Transaction, error) {
+	abi, err := bindings.ManyChainMultiSigMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	data, err := abi.Pack("execute", op, proof)
+	if err != nil {
+		return nil, err
+	}
+	tx := buildUnsignedTxFromOpts(opts, mcmsAddr, data)
+
+	return tx, nil
+}
+
+// buildSetRootCallData packs call data for ManyChainMultiSig.setRoot(...)
+func buildSetRootCallData(
+	opts *bind.TransactOpts,
+	mcmsAddr common.Address,
+	root [32]byte,
+	validUntil uint32,
+	meta bindings.ManyChainMultiSigRootMetadata,
+	metaProof [][32]byte,
+	sigs []bindings.ManyChainMultiSigSignature,
+) (*gethtypes.Transaction, error) {
+	abi, err := bindings.ManyChainMultiSigMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	data, err := abi.Pack("setRoot", root, validUntil, meta, metaProof, sigs)
+	if err != nil {
+		return nil, err
+	}
+	tx := buildUnsignedTxFromOpts(opts, mcmsAddr, data)
+
+	return tx, nil
+}
+
+func buildUnsignedTxFromOpts(
```

Comment:
Is this logic necessary? Could we just rely on the underlying provider/evm client?